src/components/editor/TextArea.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import RIEStatefulBase from 'riek/lib/RIEStatefulBase';
+export default class TextArea extends RIEStatefulBase {
+    keyDown = (event) => {
+        if (event.keyCode === 27) { this.cancelEditing() }     // Escape
+    };
+
+    renderEditingComponent = () => {
+        return <textarea
+            rows={this.props.rows}
+            cols={this.props.cols}
+            disabled={this.state.loading}
+            className={this.makeClassString()}
+            defaultValue={this.props.value}
+            onInput={this.textChanged}
+            onBlur={this.finishEditing}
+            ref="input"
+            onKeyDown={this.keyDown}
+            {...this.props.editProps} />;
+    };
+
+    renderNormalComponent = () => {
+        const value = this.state.newValue || this.props.value
+        const spans_and_brs = []
+        var anySpan = false
+        let i = 0
+        value.split("\n").map(line => {
+            if(line.length > 0){
+                anySpan = true
+                spans_and_brs.push(<span key={i}>{line}</span>)
```

Comment:
instead of keeping track of `i`, perhaps just do `spans_and_brs.length`

Reply:
Or, don't do this now, and at the end do something like `spans_and_brs.forEach( (el, i) => el.key = i)`.  Or you could do a map, but it seems like `forEach` is a little less wordy.